.docker/opt/configure-aiida.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+# This script is executed whenever the docker container is (re)started.
+
+# Debugging.
+set -x
+
+# Environment.
+reentry scan
+export SHELL=/bin/bash
+
+# Setup AiiDA autocompletion.
+grep _VERDI_COMPLETE .bash_profile &> /dev/null || echo 'eval "$(_VERDI_COMPLETE=source verdi)"' >> /home/$SYSTEM_USER/.bashrc
+
+# Check if user requested to set up AiiDA profile (and if it exists already)
+if [[ $SETUP_DEFAULT_PROFILE == true ]] && ! verdi profile show $PROFILE_NAME &> /dev/null; then
+    NEED_SETUP_PROFILE=true;
+else
+    NEED_SETUP_PROFILE=false;
+fi
+
+# Setup AiiDA profile if needed.
+if [[ $NEED_SETUP_PROFILE == true ]]; then
+
+    # Create AiiDA profile.
+    verdi quicksetup                           \
+        --non-interactive                      \
+        --profile $PROFILE_NAME                \
+        --email $USER_EMAIL                    \
+        --first-name $USER_FIRST_NAME          \
+        --last-name $USER_LAST_NAME            \
+        --institution $USER_INSTITUTION        \
+        --db-backend $AIIDADB_BACKEND
+
+    # Setup and configure local computer.
+    computer_name=localhost
+    verdi computer show $computer_name || verdi computer setup \
+        --non-interactive                                      \
+        --label ${computer_name}                               \
+        --description "this computer"                          \
+        --hostname ${computer_name}                            \
+        --transport local                                      \
+        --scheduler direct                                     \
+        --work-dir /home/aiida/aiida_run/                      \
+        --mpirun-command "mpirun -np {tot_num_mpiprocs}"       \
+        --mpiprocs-per-machine 1 &&                            \
+    verdi computer configure local ${computer_name}            \
+        --non-interactive                                      \
+      --safe-interval 0.0
+fi
+
+# Migration is run if at least one profile exists.
+# Daemon is run if the database is either up to date or it has been successfully migrated.
+( verdi profile show && verdi database migrate --force && verdi daemon start ) || echo "AiiDA daemon is not running."
```

Comment:
I am not sure if it is a good idea to chain all of these commands. If the first or second fails, it will still show that the daemon is not running. While technically correct, it is a bit misleading. Why not separate the three each which its own error message?

Reply:
I think you are right. Migration won't run anyways if the default profile is not provided. Same true for the `daemon start`- it will work only if (a) default profile exists and (b) it does not need migration.

Reply:
done